### PR TITLE
Remove folder settings tab, move folder options to properties config

### PR DESCRIPTION
**Pull request by DanVanAtta**
Benefits:
- code configuration is simpler. Only complication is defaulting when a specified folder does not exist.
- less UI, good for users
- Removes the need for map folder param. That is used for bots to specify where the map folder is (so it can be shared). If we can specify that in config now, then we can just do that and do not need the command line arg.

Settings screen after (note there is no folders tab):
no_folders

### CODE REVIEW

**Comment by ron-murhammer** [comment]
@DanVanAtta I actually kind of like the feature of the ability to set save game and maps folder. I don't know that removing that from settings is such a good idea. Unless we have feedback that very few users care about the options or use it then I think we should keep it.

File: src/main/java/games/strategy/debug/ClientLogger.java [hunk 1]
```diff
@@ -47,4 +47,15 @@ public static void logError(final String msg, final Collection<? extends Throwab
     logError(msg);
     throwables.forEach(ClientLogger::logError);
   }
+
+  public static FatalErrorException fatal(final String error) {
+    logError("Fatal error: " + error);
+    return new FatalErrorException(error);
+  }
+
+  public static class FatalErrorException extends RuntimeException {
```
Comment (ssoloff) [inline src/main/java/games/strategy/debug/ClientLogger.java]: NBD, but... Static analysis: "The serializable class FatalErrorException does not declare a static final serialVersionUID field of type long"

File: src/main/java/games/strategy/engine/config/GameEngineProperty.java [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+  ENGINE_VERSION("engine_version"),
+  LOBBY_PROPS_URL("lobby_properties_file_url"),
+  LOBBY_PROPS_BACKUP_FILE("lobby_properties_file_backup"),
+  MAP_FOLDER("map_folder"),
+  SAVED_GAMES_FOLDER("saved_games_folder");
 
   private final String value;
 
-  private GameEngineProperty(final String value) {
+  GameEngineProperty(final String value) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/config/GameEngineProperty.java]: Was the accessibility increased for test purposes?  `@VisibleForTesting`?  However, I don't see any usage outside this class.

**Comment by DanVanAtta** [comment]
@ron-murhammer Interesting feedback, glad you've found the folder settings useful. . I'll note that the feature is not going away completely, instead the path will be in 'game_engine.properties': https://github.com/triplea-game/triplea/pull/1843/files#diff-51766a8de1f4e3121dd69af1c0a5a5baR14

File: src/main/java/games/strategy/engine/config/GameEnginePropertyFileReader.java [hunk 3]
```diff
@@ -24,21 +26,33 @@ public GameEnginePropertyFileReader() {
...
+  }
+
+  private String readProperty(final String propertyKey) {
+
     try (FileInputStream inputStream = new FileInputStream(propertyFile)) {
       final Properties props = new Properties();
       props.load(inputStream);
 
-      if (!props.containsKey(propertyKey.toString())) {
-        throw new PropertyNotFoundException(propertyKey);
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/config/GameEnginePropertyFileReader.java]: Looks like `PropertyNotFoundException` has no more uses.  Can it be removed?

**Comment by ron-murhammer** [comment]
@DanVanAtta Understood. I think the question is do many users find it useful as if we move it to a property file then very few people will ever know its there or understand how to change it.

File: src/main/java/games/strategy/triplea/settings/folders/FolderSettings.java [hunk 4]
```diff
@@ -1,51 +1,97 @@
...
+      Preconditions.checkState(newFolder.isDirectory(), newFolder.getAbsolutePath() + " should be a directory");
+      return newFolder;
+    }
+
+    if (!fileMaker.apply(newFolder)) {
+      // failed to create, try the default
+      newFolder = new File(ClientFileSystemHelper.getUserRootFolder(), defaultValue);
+      ClientLogger.logError(
+          "Failed to create folder: " + newFolder.getAbsolutePath()
+              + ", falling back to default: " + newFolder.getAbsolutePath());
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/settings/folders/FolderSettings.java]: You are logging `newFolder` twice, but my impression is that the first instance should refer to the folder that could not be created when applying `fileMaker`.  Might need a second local variable here so you can capture both folders in the log message.

**Comment by DanVanAtta** [comment]
- Let's keep the folder setting, it seems like it is in use after all!
- The command line arg for map folder should IMO go to a config, to support the bot case. We'll still keep the UI setting, but allow for a properties override value. That is accomplished in this branch, but we also go too far here as well (much too far).

**Pull request closed without merging**